services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/document/DefaultEnhancedDocument.java

```diff
@@ -0,0 +1,491 @@
+ * internally saves attributeValueMap which is saved to ddb as it is. The aattribute values are retrieved
+ * by converting attributeValue from attributeValueMap at the time of get.
+ */
+@Immutable
+@SdkInternalApi
+public class DefaultEnhancedDocument implements EnhancedDocument {
+
+    private final Map<String, AttributeValue> attributeValueMap;
+
+    private final ChainConverterProvider attributeConverterProviders;
+
+    public DefaultEnhancedDocument(Map<String, AttributeValue> attributeValueMap,
+                                   ChainConverterProvider attributeConverterProviders) {
+        this.attributeValueMap = attributeValueMap;
+        this.attributeConverterProviders = attributeConverterProviders;
+    }
+
+    public DefaultEnhancedDocument(DefaultBuilder builder) {
+        attributeValueMap = Collections.unmodifiableMap(builder.getAttributeValueMap());
+        attributeConverterProviders = ChainConverterProvider.create(builder.attributeConverterProviders);
+    }
+
+    public static DefaultBuilder builder() {
+        return new DefaultBuilder();
+
+    }
+
+    @Override
+    public Builder toBuilder() {
+        return new DefaultBuilder(this);
+    }
+
+    public Map<String, AttributeValue> getAttributeValueMap() {
+        return attributeValueMap;
+    }
+
+    @Override
+    public boolean isNull(String attributeName) {
+        return isPresent(attributeName) && attributeValueMap.get(attributeName).nul();
+    }
+
+    @Override
+    public boolean isPresent(String attributeName) {
+        return attributeValueMap.containsKey(attributeName);
+    }
+
+    @Override
+    public <T> T get(String attributeName, EnhancedType<T> type) {
+        AttributeConverter<T> attributeConverter = attributeConverterProviders.converterFor(type);
+        if (attributeConverter == null) {
+            throw new IllegalArgumentException("type " + type + " is not found in AttributeConverterProviders");
+        }
+        AttributeValue attributeValue = attributeValueMap.get(attributeName);
+        if (attributeValue == null) {
+            return null;
+        }
+        return attributeConverter.transformTo(attributeValue);
+    }
+
+    @Override
+    public String getString(String attributeName) {
+        AttributeValue attributeValue = attributeValueMap.get(attributeName);
+        return attributeValue != null
+               ? attributeConverterProviders.converterFor(EnhancedType.of(String.class)).transformTo(attributeValue)
+               : null;
+    }
+
+    @Override
+    public SdkNumber getSdkNumber(String attributeName) {
+        AttributeValue attributeValue = attributeValueMap.get(attributeName);
+        return attributeValue != null
+               ? SdkNumber.fromString(
+            attributeConverterProviders.converterFor(EnhancedType.of(String.class)).transformTo(attributeValue))
+               : null;
```

Comment:
Nit: Can this be formatted with line breaks that are easier to read, ie not `SdkNumber.fromString(`